compiler/src/posix.mak

```diff
@@ -1,217 +0,0 @@
-# AUTO_BOOTSTRAP:       Enable auto-boostrapping by downloading a stable DMD binary
-# INSTALL_DIR:          Installation folder to use
-# MODEL:                Target architecture to build for (32,64) - defaults to the host architecture
-#
-################################################################################
-# Build modes:
-# ------------
-# BUILD: release (default) | debug (enabled a build with debug instructions)
-#
-# Opt-in build features:
-#
-# ENABLE_RELEASE:       Optimized release build
-# ENABLE_DEBUG:         Add debug instructions and symbols (set if ENABLE_RELEASE isn't set)
-# ENABLE_ASSERTS:       Don't use -release if ENABLE_RELEASE is set
-# ENABLE_LTO:           Enable link-time optimizations
-# ENABLE_UNITTEST:      Build dmd with unittests (sets ENABLE_COVERAGE=1)
-# ENABLE_PROFILE:       Build dmd with a profiling recorder (D)
-# ENABLE_COVERAGE       Build dmd with coverage counting
-# ENABLE_SANITIZERS     Build dmd with sanitizer (e.g. ENABLE_SANITIZERS=address,undefined)
-#
-# Targets
-# -------
-#
-# all					Build dmd
-# unittest              Run all unittest blocks
-# cxx-unittest          Check conformance of the C++ headers
-# build-examples        Build DMD as library examples
-# clean                 Remove all generated files
-# man                   Generate the man pages
-# checkwhitespace       Checks for trailing whitespace and tabs
-# zip                   Packs all sources into a ZIP archive
-# gitzip                Packs all sources into a ZIP archive
-# install               Installs dmd into $(INSTALL_DIR)
-################################################################################
-
-$(warning ===== DEPRECATION NOTICE ===== )
-$(warning ===== DEPRECATION: posix.mak is deprecated. Please use src/build.d instead.)
-$(warning ============================== )
-
-BUILD_JOBS_ARG=
-ifneq (,$(BUILD_JOBS))
-  BUILD_JOBS_ARG = -j$(BUILD_JOBS)
-endif
-
-# Forward D compiler bootstrapping to bootstrap.sh
-ifneq (,$(AUTO_BOOTSTRAP))
-default:
-	@bash ./bootstrap.sh $(BUILD_JOBS_ARG)
-.DEFAULT:
-	@bash ./bootstrap.sh "$@" $(BUILD_JOBS_ARG)
-else
-
-# get OS and MODEL
-include osmodel.mak
-
-# Default to a release built, override with BUILD=debug
-ifeq (,$(BUILD))
-BUILD=release
-endif
-
-ifneq ($(BUILD),release)
-    ifneq ($(BUILD),debug)
-        $(error Unrecognized BUILD=$(BUILD), must be 'debug' or 'release')
-    endif
-endif
-
-INSTALL_DIR=../../install
-D = dmd
-
-GENERATED = ../../generated
-G = $(GENERATED)/$(OS)/$(BUILD)/$(MODEL)
-$(shell mkdir -p $G)
-
-ifeq (osx,$(OS))
-    export MACOSX_DEPLOYMENT_TARGET=10.9
-endif
-
-HOST_CXX?=c++
-# compatibility with old behavior
-ifneq ($(HOST_CC),)
-  $(warning ===== WARNING: Please use HOST_CXX=$(HOST_CC) instead of HOST_CC=$(HOST_CC). =====)
-  HOST_CXX=$(HOST_CC)
-endif
-
-HOST_DC?=
-ifneq (,$(HOST_DC))
-  $(warning ========== Use HOST_DMD instead of HOST_DC ========== )
-  HOST_DMD=$(HOST_DC)
-endif
```

Comment:
Possibly this is what's missing in the makefile.

Reply:
Oh right, thx for looking! I thought `DAutoTest` was using the `compiler/src/posix.mak` (removed without proxy), but no, it's the `src/posix.mak` one.

Reply:
Oh well, it does use the removed one sometime later apparently:
```
make --directory=../dmd/compiler/src -f posix.mak AUTO_BOOTSTRAP=1 BUILD_JOBS=
make[1]: Entering directory '/tmpfs/dautotest-work/repo/dmd/compiler/src'
make[1]: posix.mak: No such file or directory
```

Reply:
I suspect that is the dlang.org repo though.

https://github.com/dlang/dlang.org/blob/master/posix.mak#L596-L600